compiler/semtypes.nim

```diff
@@ -1009,8 +1009,11 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
       result.sons[0] = r
       result.n.typ = r
 
-  if genericParams != nil:
+  if genericParams != nil and genericParams.len > 0:
     for n in genericParams:
+      if {sfUsed, sfAnon} * n.sym.flags == {}:
```

Comment:
For the particular purpose here, `sfUsed` seemed like an easy solution. The type params in question are created shortly before this check (in the body of `semProcTypeNode`) and they are marked as `sfUsed` during the signature processing. I can check how much the user can interfere with this process. One alternative would be to introduce an additional "internal use" flag.